From a69237c389f6bb6e7a5e0bf53e4e77fc114be9a3 Mon Sep 17 00:00:00 2001
From: Dien Pham <dien.pham.ry@rvc.renesas.com>
Date: Fri, 27 May 2016 16:36:54 +0700
Subject: [PATCH 026/909] soc: renesas: rcar-sysc: Add helper function to get
 domain name

commit 38adb27f66c2af6d481876d87c5e1e41586946d5 from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git

This function will get the name power domain that contains sysc_ch.
It is intended to be used for next update times.

Signed-off-by: Dien Pham <dien.pham.ry@rvc.renesas.com>
(cherry picked from commit 6ac8938a540df33fffc505700610665a3c541cb5)
Signed-off-by: Takeshi Kihara <takeshi.kihara.df@renesas.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/soc/renesas/rcar-sysc.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/soc/renesas/rcar-sysc.c b/drivers/soc/renesas/rcar-sysc.c
index b09bd0fcb8f4..d78d2ca72fe6 100644
--- a/drivers/soc/renesas/rcar-sysc.c
+++ b/drivers/soc/renesas/rcar-sysc.c
@@ -61,6 +61,9 @@
 static void __iomem *rcar_sysc_base;
 static DEFINE_SPINLOCK(rcar_sysc_lock); /* SMP CPUs + I/O devices */
 
+
+static const char *to_pd_name(const struct rcar_sysc_ch *sysc_ch);
+
 static int rcar_sysc_pwr_on_off(const struct rcar_sysc_ch *sysc_ch, bool on)
 {
 	unsigned int sr_bit, reg_offs;
@@ -176,6 +179,11 @@ static inline struct rcar_sysc_pd *to_rcar_pd(struct generic_pm_domain *d)
 	return container_of(d, struct rcar_sysc_pd, genpd);
 }
 
+static inline const char *to_pd_name(const struct rcar_sysc_ch *sysc_ch)
+{
+	return container_of(sysc_ch, struct rcar_sysc_pd, ch)->genpd.name;
+}
+
 static int rcar_sysc_pd_power_off(struct generic_pm_domain *genpd)
 {
 	struct rcar_sysc_pd *pd = to_rcar_pd(genpd);
-- 
2.17.1

